packages/go_router/lib/src/delegate.dart

```diff
@@ -59,13 +61,12 @@ class GoRouterDelegate extends RouterDelegate<RouteMatchList>
       final RouteBase route = match.route;
 
       if (route is GoRoute && route.parentNavigatorKey != null) {
-        // It should not be possible for a GoRoute with parentNavigatorKey to be
-        // the only page, so maybePop should never return false in this case.
-        assert(await route.parentNavigatorKey!.currentState!.maybePop());
```

Comment:
oh I must have missed this, the maybepop should not be in the assert, only the canPop should be in the assert.
I imagine the code should be the following.

```dart
final bool maybePop = await route.parentNavigatorKey!.currentState!.maybePop();
assert(maybePop);
return maybePop;

Reply:
It is ok to route.parentNavigatorKey!.currentState!.canPop();
in the assert because calling it will not get any side effect, but maybePop can't because it will have side effect.

Reply:
We shouldn't assert that maybePop is true, because parentNavigatorKey can be defined on the first GoRoute , see the original issue ([#111861](https://github.com/flutter/flutter/issues/111861)).

```dart
  final GoRouter _router = GoRouter(
    navigatorKey: _rootNavigatorKey,
    initialLocation: '/c',
    routes: <RouteBase>[
      GoRoute(
        parentNavigatorKey: _rootNavigatorKey,
        path: '/c',
        builder: (BuildContext context, GoRouterState state) {
          return const ScreenC();
        },
      ),

```

Reply:
sounds good